src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufSerializer.cs

```diff
@@ -0,0 +1,373 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources;
+
+namespace OpenTelemetry.Exporter.Geneva;
+
+internal class OtlpProtobufSerializer
+{
+    private const int LengthAndTagSize = 4;
+
+    private static readonly ConcurrentBag<List<Metric>> MetricListPool = new();
+
+    private readonly Dictionary<string, List<Metric>> scopeMetrics = new();
+
+    private int resourceMetricStartIndex;
+
+    private int previousScopeMetricsStartIndex;
+
+    private int previousMetricStartIndex;
+
+    private int previousIntrumentStartIndex;
+
+    private int previousMetricPointStartIndex;
+
+    internal IMetricDataTransport MetricDataTransport;
+
+    public OtlpProtobufSerializer(IMetricDataTransport metricDataTransport)
+    {
+        this.MetricDataTransport = metricDataTransport;
+    }
+
+    internal void SerializeAndSendMetrics(byte[] buffer, ref int currentPosition, Resource resource, in Batch<Metric> metricBatch)
+    {
+        foreach (var metric in metricBatch)
+        {
+            if (this.scopeMetrics.TryGetValue(metric.MeterName, out var metricList))
+            {
+                metricList.Add(metric);
+            }
+            else
+            {
+                if (MetricListPool.TryTake(out var newList))
```

Comment:
Why do we need to pool the lists? We can simply create a dedicated list for each meter and clear each of them at the end of export.